pkg/console/subresource/deployment/deployment.go

```diff
@@ -358,23 +358,39 @@ func livenessProbe() *corev1.Probe {
 // one ready replica
 func IsReady(deployment *appsv1.Deployment) bool {
 	avail := deployment.Status.ReadyReplicas >= 1
-	if avail {
-		klog.V(4).Infof("deployment is available, ready replicas: %v", deployment.Status.ReadyReplicas)
-	} else {
+	if !avail {
 		klog.V(4).Infof("deployment is not available, ready replicas: %v", deployment.Status.ReadyReplicas)
 	}
 	return avail
 }
 
 func IsReadyAndUpdated(deployment *appsv1.Deployment) bool {
-	return deployment.Status.Replicas == deployment.Status.ReadyReplicas &&
-		deployment.Status.Replicas == deployment.Status.UpdatedReplicas
+	ready := deployment.Status.Replicas == deployment.Status.ReadyReplicas
+	updated := deployment.Status.Replicas == deployment.Status.UpdatedReplicas
+	if !ready {
+		klog.V(4).Infof("deployment is not ready, ready replicas: %v/%v", deployment.Status.ReadyReplicas, deployment.Status.Replicas)
+	}
+	if !updated {
+		klog.V(4).Infof("deployment has not updated, updated replicas: %v/%v", deployment.Status.UpdatedReplicas, deployment.Status.Replicas)
```

Comment:
I think you want to compare updated replicas to spec replicas

Reply:
I'll update.